integration-test/src/test/java/org/ehcache/integration/ThreadPoolsTest.java

```diff
@@ -102,4 +104,23 @@ public void testDefaultThreadPoolFailsWithExplicitConfig() throws Exception {
     cacheManager.close();
   }
 
+  @Test
+  public void testNoDefaultThreadPoolFailsWithNoCacheEventDispatcher() throws Exception {
+    PooledExecutionServiceConfiguration executionServiceConfiguration = new PooledExecutionServiceConfiguration();
+    executionServiceConfiguration.addPool("foo", 2, 4);
+    CacheManager cacheManager = CacheManagerBuilder.newCacheManagerBuilder()
```

Comment:
The pre-existing code here probably predates try-with-resources, but that shouldn't stop you using it in new code.